### ISSUE TITLE
I always have timeout error callback called after the location search success

### ISSUE THREAD

**Issue by nmDejamobile** [malcommac/SwiftLocation#53]
I run Location like this :

```
let request: SwiftLocation.Request = SwiftLocation.Location.getLocation(withAccuracy: .Block, frequency: .ByDistanceIntervals(meters: 100), timeout: 10, onSuccess: { (location) in
    debugPrint(NSDate().timeIntervalSince1970 * 1000, "Location acquired")
}) { (lastValidLocation, error) in
    debugPrint(NSDate().timeIntervalSince1970 * 1000, "Location error", error._code, error.description)
}
print("Start location request \(request!.UUID)\n")
request.start()
```

and both events are triggered :

```
Start location request DEBCACED-1C5E-4E0D-B7E0-3DB9D273C9DA

1471859016543.7268 "Location acquired"
1471859026548.197 "Location error" 1 "Timeout for request"
```

So transaction is canceled and then no further location change triggers the action.

**Comment by nmDejamobile** [comment]
Note that setTimeoutTimer is always set (excepted when Location location is not valid).
So even for .OneShot, timeout is started after the request is canceled.
I think that something could be done here : https://github.com/malcommac/SwiftLocation/blob/master/src/LocationRequest.swift#L195

**Comment by nmDejamobile** [comment]
FIxed by PR #54
